tests/lib/test_config.py

```diff
@@ -1,8 +1,87 @@
 # pylint: disable=missing-function-docstring disable=missing-module-docstring
 
+from unittest.mock import call
+
+import pytest
+
+from nestor_api.config.config import Configuration
 import nestor_api.lib.config as config
+from nestor_api.lib.errors import AppConfigurationNotFoundError
+import nestor_api.lib.io as io
+
+
+def test_change_environment(mocker):
+    spy = mocker.patch.object(io, "execute")
+
+    config.change_environment("environment", "path/to/config")
+
+    spy.assert_has_calls(
+        [
+            call("git stash", "path/to/config"),
+            call("git fetch origin", "path/to/config"),
+            call("git checkout environment", "path/to/config"),
+            call("git reset --hard origin/environment", "path/to/config"),
+        ]
+    )
+
+
+@pytest.mark.usefixtures("config")
+def test_create_temporary_directory(mocker):
+    spy = mocker.patch.object(io, "create_temporary_copy")
+
+    config.create_temporary_config_copy()
+
+    spy.assert_called_once_with("/fixtures-nestor-config", "config")
```

Comment:
Maybe assert on the return value because it is almost as important as the rest, otherwise you cannot guess where the temporary copy has been created 🙂